pkg/controller/container-runtime-config/helpers_test.go

```diff
@@ -406,3 +407,100 @@ func TestValidateRegistriesConfScopes(t *testing.T) {
+				},
+			},
+			expectedRegistriesBlocked: []string{"block.io", "block-2.io"},
+			expectedPolicyBlocked:     []string{"block.io", "block-2.io"},
+			expectedErr:               false,
+		},
+		{
+			name:       "regular blocked list with unrelated mirror rules configured",
+			releaseImg: "payload-reg.io/release-image:v4.0",
+			imgSpec: &apicfgv1.ImageSpec{
+				RegistrySources: apicfgv1.RegistrySources{
+					BlockedRegistries: []string{"block.io", "block-2.io"},
+				},
+			},
+			icspRules: []*apioperatorsv1alpha1.ImageContentSourcePolicy{
+				{
+					Spec: apioperatorsv1alpha1.ImageContentSourcePolicySpec{
+						RepositoryDigestMirrors: []apioperatorsv1alpha1.RepositoryDigestMirrors{
+							{Source: "src.io/payload", Mirrors: []string{"mirror-1.io/payload", "mirror-2.io/payload"}},
+						},
+					},
+				},
+			},
+			expectedRegistriesBlocked: []string{"block.io", "block-2.io"},
+			expectedPolicyBlocked:     []string{"block.io", "block-2.io"},
+			expectedErr:               false,
+		},
+		{
+			name:       "payload reg does not have mirror configured and is in blocked list",
+			releaseImg: "payload-reg.io/release-image:v4.0",
+			imgSpec: &apicfgv1.ImageSpec{
+				RegistrySources: apicfgv1.RegistrySources{
+					BlockedRegistries: []string{"block.io", "payload-reg.io", "block-2.io"},
+				},
+			},
+			icspRules: []*apioperatorsv1alpha1.ImageContentSourcePolicy{
+				{
+					Spec: apioperatorsv1alpha1.ImageContentSourcePolicySpec{
+						RepositoryDigestMirrors: []apioperatorsv1alpha1.RepositoryDigestMirrors{
+							{Source: "src.io/payload", Mirrors: []string{"mirror-1.io/payload", "mirror-2.io/payload"}},
+						},
+					},
+				},
+			},
+			expectedRegistriesBlocked: []string{"block.io", "block-2.io"},
+			expectedPolicyBlocked:     []string{"block.io", "block-2.io"},
+			expectedErr:               true,
+		},
+		{
+			name:       "payload reg has mirror configured and is in blocked list",
+			releaseImg: "payload-reg.io/release-image:v4.0",
+			imgSpec: &apicfgv1.ImageSpec{
+				RegistrySources: apicfgv1.RegistrySources{
+					BlockedRegistries: []string{"block.io", "payload-reg.io", "block-2.io"},
+				},
+			},
+			icspRules: []*apioperatorsv1alpha1.ImageContentSourcePolicy{
+				{
+					Spec: apioperatorsv1alpha1.ImageContentSourcePolicySpec{
+						RepositoryDigestMirrors: []apioperatorsv1alpha1.RepositoryDigestMirrors{
+							{Source: "payload-reg.io/release-image", Mirrors: []string{"mirror-1.io/payload", "mirror-2.io/payload"}},
+						},
+					},
+				},
+			},
+			expectedRegistriesBlocked: []string{"block.io", "payload-reg.io", "block-2.io"},
+			expectedPolicyBlocked:     []string{"block.io", "block-2.io"},
+			expectedErr:               false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotRegistries, gotPolicy, err := getValidBlockedRegistries(tt.releaseImg, tt.imgSpec, tt.icspRules)
+			if err != nil && !tt.expectedErr {
```

Comment:
This should also fail on `tt.expectedErr && err == nil`.

Reply:
fixed